streaming/python/ray/streaming/runtime/queue/queue_interface.py

```diff
@@ -0,0 +1,203 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from abc import ABCMeta, abstractmethod
+
+import ray
+import ray.streaming.runtime.queue.queue_utils as queue_utils
+
+
+class QueueID:
+    """
+    QueueID is used to identify a queue which is the transfer channel between
+     a upstream operator and downstream operator.
+    """
+    def __init__(self, str_qid: str):
+        """
+        :param str_qid: string representation of queue id
+        """
+        self.str_qid = str_qid
+        self.object_qid = ray.ObjectID(queue_utils.qid_str_to_bytes(str_qid))
+
+    def __eq__(self, other):
+        if other is None:
+            return False
+        if type(other) is QueueID:
+            return self.str_qid == other.str_qid
+        else:
+            return False
+
+    def __hash__(self):
+        return hash(self.str_qid)
+
+    def __repr__(self):
+        return self.str_qid
+
+
+class QueueItem:
+    """
+    QueueItem interface lists all methods that a QueueItem subclass need to implement.
+    QueueItem represents an item pulled from queue. It maybe a data message or control message.
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self):
+        pass
+
+    @abstractmethod
+    def body(self):
+        """
+        Queue item data
+        """
+        pass
+
+    @abstractmethod
+    def timestamp(self):
+        """
+        Timestamp when item is written by upstream StreamingWriter
+        :return:
+        """
+        pass
+
+
+class QueueMessage(QueueItem):
+    """
+    Queue data message interface
+    """
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def queue_id(self):
+        """
+        :return: id of queue where data is coming from
+        """
+        pass
+
+
+class QueueLink:
+    """
+    queue link interface
+    """
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def set_ray_runtime(self, runtime):
+        """
+        Set ray runtime config
+        :param runtime:  ray runtime config
+        """
+        pass
+
+    @abstractmethod
+    def set_configuration(self, conf):
+        """
+        Set queue configuration
+        :param conf:  queue configuration
+        """
+        pass
+
+    @abstractmethod
+    def register_queue_consumer(self, input_queue_ids, from_actors):
+        """
+        Get queue consumer of input queues
+        :param input_queue_ids:  input queue ids
+        :param from_actors:  upstream input actors
+        :return:  queue consumer
```

Comment:
Can you change  docstrings to follow google python style? E.g., this one should be
```
"""Get queue consumer of input queues.

Args:
    input_queue_ids:  Input queue ids.
    from_actors: Upstream input actors.

Returns:
    Queue consumer.
"""
```

BTW, this method is named `register_queue_consumer`, but doc says "get".

Reply:
fixed comments, `QueueLink` will be removed soon, thus `register_queue_consumer/register_queue_producer` will exists any more

Reply:
fixed